mithril-stm/src/schnorr_signature/signature.rs

```diff
@@ -0,0 +1,206 @@
+    /// Convert an `SchnorrSignature` into bytes.
+    pub fn to_bytes(self) -> [u8; 96] {
+        let mut out = [0; 96];
+        out[0..32].copy_from_slice(&self.sigma.to_bytes());
+        out[32..64].copy_from_slice(&self.signature.to_bytes());
+        out[64..96].copy_from_slice(&self.challenge.to_bytes());
+
+        out
+    }
+
+    /// Convert bytes into a `SchnorrSignature`.
+    pub fn from_bytes(bytes: &[u8]) -> StmResult<Self> {
+        if bytes.len() < 96 {
+            return Err(anyhow!(SchnorrSignatureError::SerializationError))
+                .with_context(|| "Not enough bytes provided to create a signature.");
+        }
+
+        let sigma_bytes = bytes[0..32]
+            .try_into()
+            .map_err(|_| anyhow!(SchnorrSignatureError::SerializationError))
+            .with_context(|| "Failed to obtain sigma's bytes.")?;
+        let sigma = JubjubExtended::from_bytes(&sigma_bytes)
+            .into_option()
+            .ok_or(anyhow!(SchnorrSignatureError::SerializationError))
+            .with_context(|| "Unable to convert bytes into a sigma value.")?;
+
+        let signature_bytes = bytes[32..64]
+            .try_into()
+            .map_err(|_| anyhow!(SchnorrSignatureError::SerializationError))
+            .with_context(|| "Failed to obtain signature's bytes.")?;
+        let signature = JubjubScalar::from_bytes(&signature_bytes)
+            .into_option()
+            .ok_or(anyhow!(SchnorrSignatureError::SerializationError))
+            .with_context(|| "Unable to convert bytes into a signature value.")?;
+
+        let challenge_bytes = bytes[64..96]
+            .try_into()
+            .map_err(|_| anyhow!(SchnorrSignatureError::SerializationError))
+            .with_context(|| "Failed to obtain challenge's bytes.")?;
+        let challenge = JubjubScalar::from_bytes(&challenge_bytes)
+            .into_option()
+            .ok_or(anyhow!(SchnorrSignatureError::SerializationError))
+            .with_context(|| "Unable to convert bytes into a challenge value.")?;
+
+        Ok(Self {
+            sigma,
+            signature,
+            challenge,
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use crate::{SchnorrSignature, SchnorrSigningKey};
+    use rand_chacha::ChaCha20Rng;
+    use rand_core::SeedableRng;
+
+    #[test]
+    fn serialize_deserialize_signature() {
+        let mut rng = ChaCha20Rng::from_seed([0u8; 32]);
+
+        let msg = vec![0, 0, 0, 1];
+        let sk = SchnorrSigningKey::generate(&mut rng).unwrap();
+
+        let sig = sk.sign(&msg, &mut rng).unwrap();
+        let sig_bytes: [u8; 96] = sig.to_bytes();
+        let sig2 = SchnorrSignature::from_bytes(&sig_bytes).unwrap();
+
+        assert_eq!(sig, sig2);
+    }
+
+    #[test]
+    fn from_bytes_signature_not_enough_bytes() {
+        let msg = vec![0u8; 95];
+
+        let result = SchnorrSignature::from_bytes(&msg);
+
+        assert!(result.is_err());
+    }
+
+    const GOLDEN_BYTES: &[u8; 96] = &[
```

Comment:
Could you put the golden tests in a `golden` sub-module for readability and maintainability?